Content.Server/Abilities/Mime/MimePowersSystem.cs

```diff
@@ -0,0 +1,144 @@
+using Content.Server.Popups;
+using Content.Server.Coordinates.Helpers;
+using Content.Shared.Speech;
+using Content.Shared.Actions;
+using Content.Shared.Alert;
+using Content.Shared.Maps;
+using Content.Shared.MobState.Components;
+using Content.Shared.Tag;
+using Content.Shared.Doors.Components;
+using Robust.Shared.Player;
+
+namespace Content.Server.Abilities.Mime
+{
+    public sealed class MimePowersSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+        [Dependency] private readonly AlertsSystem _alertsSystem = default!;
+        [Dependency] private readonly TagSystem _tagSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<MimePowersComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<MimePowersComponent, SpeakAttemptEvent>(OnSpeakAttempt);
+            SubscribeLocalEvent<MimePowersComponent, InvisibleWallActionEvent>(OnInvisibleWall);
+        }
+
+        private Queue<EntityUid> RemQueue = new();
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+            foreach (var entity in RemQueue)
+            {
+                RemComp<VowbreakerComponent>(entity);
+            }
+            RemQueue.Clear();
+            /// Queue to despawn invis walls
+            foreach (var invisWall in EntityQuery<InvisibleWallComponent>())
+            {
+                invisWall.Accumulator += frameTime;
+                if (invisWall.Accumulator < invisWall.DespawnTime)
+                {
+                    continue;
+                }
+                EntityManager.QueueDeleteEntity(invisWall.Owner);
+            }
+            /// Queue to track whether mimes can retake vows yet
+            foreach (var (vowbreaker, mime) in EntityQuery<VowbreakerComponent, MimePowersComponent>())
```

Comment:
if the mime powers component doesn't get removed when the vow is broken, then the vowbreaker comp shouldnt be separate at all imo.

Reply:
either:
1) turn the mime actions into action prototypes, remove `MimePowersComponent` (and add vowbreaker) when the vow is broken, remove the `Enabled` field from `MimePowersComponent` and readd `MimePowersComponent` when they take the vow again
2) merge vowbreaker fields into mime powers, remove the queue stuff